### PR TITLE
src/bundle: download temporary bundle to a separate temp directory

### PR DESCRIPTION
**Pull request by jluebbe**
This protects against the issue described in #819.

### CODE REVIEW

File: src/bundle.c [hunk 1]
```diff
@@ -1412,7 +1412,7 @@ gboolean check_bundle(const gchar *bundlename, RaucBundle **bundle, gboolean ver
 	if (is_remote_scheme(bundlescheme)) {
 #if ENABLE_NETWORK
 		ibundle->origpath = g_strdup(bundlename);
-		ibundle->path = g_build_filename(g_get_tmp_dir(), "_download.raucb", NULL);
+		ibundle->path = g_build_filename(g_dir_make_tmp("rauc-XXXXXX", NULL), "download.raucb", NULL);
```
Comment (a3f) [inline src/bundle.c:1415]: Leaks.

**Review by a3f — CHANGES_REQUESTED** [review]

File: src/bundle.c (hunk 1 above)
Comment (a3f) [reply src/bundle.c:1415]: And while at it, you may want to check for `NULL` and return, even if the error is unlikely.

**Review by a3f — DISMISSED** [review]
LGTM. Nitpick: Code always sets `res = FALSE` in error case, even if res hasn't been changed from initial value. (See `if (verify` branch above). It might be good to follow this scheme. Makes code more robust against future refactoring.

**Review by a3f — DISMISSED** [review]

**Comment by ejoerns** [comment]
With this change, the tmdir is created but (unlike the bundle itself) not removed after installation/error.

**Review by ejoerns — APPROVED** [review]
Code changes look fine for me now.

**Review by ejoerns — APPROVED** [review]

**Pull request merged**